src/components/SolutionListItem.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useEffect, useState } from "react";
+import RenderElement from "./RenderElement";
+import { contentTypes } from "../model/project";
+import { articleLink } from "../constants/links";
+import { createClient } from "../utils/client";
+import { useAppContext } from "../context/AppContext";
+import { Solution } from "../model/content-types/solution";
+import { Replace } from "../utils/types";
+import { DeliveryError } from "@kontent-ai/delivery-sdk";
+
+const SolutionList: React.FC = () => {
+  const { environmentId, apiKey } = useAppContext();
+
+  const [solutions, setSolutions] = useState<Solution[] | null>(null);
```

Comment:
The array type should be readonly as it is forbidden to modify anything inside a react state.